src/pgrn-custom-scan.c

```diff
@@ -570,6 +610,48 @@ PGrnSearchBuildCustomScanConditions(CustomScanState *customScanState,
 	}
 }
 
+static void
+PGrnCustomScanSort(CustomScanState *customScanState)
+{
+	PGrnScanState *state = (PGrnScanState *) customScanState;
+	Relation table = customScanState->ss.ss_currentRelation;
+	int nSortKeys = list_length(state->pathKeys);
+	grn_table_sort_key *sortKeys =
+		(grn_table_sort_key *) palloc(sizeof(grn_table_sort_key) * nSortKeys);
+	ListCell *cell;
+	unsigned int i = 0;
+	foreach (cell, state->pathKeys)
+	{
+		PathKey *pathKey = (PathKey *) lfirst(cell);
+		EquivalenceMember *member = linitial(pathKey->pk_eclass->ec_members);
+		Expr *expr = (Expr *) (member->em_expr);
+		if (IsA(expr, Var))
+		{
+			// Support only simple sorting by columns.
+			Var *var = (Var *) expr;
+			Form_pg_attribute attr =
+				TupleDescAttr(table->rd_att, var->varattno - 1);
+			const char *name = NameStr(attr->attname);
+			sortKeys[i].key =
+				grn_obj_column(ctx, state->searched, name, strlen(name));
+			if (pathKey->pk_cmptype == COMPARE_LT)
+				sortKeys[i].flags = GRN_TABLE_SORT_ASC;
+			else if (pathKey->pk_cmptype == COMPARE_GT)
+				sortKeys[i].flags = GRN_TABLE_SORT_DESC;
+			i++;
+		}
+	}
+
+	state->sorted = grn_table_create(
+		ctx, NULL, 0, NULL, GRN_OBJ_TABLE_NO_KEY, NULL, state->searched);
+	grn_table_sort(
+		ctx, state->searched, 0, -1, state->sorted, sortKeys, nSortKeys);
+
+	for (i = 0; i < nSortKeys; i++)
+		grn_obj_close(ctx, sortKeys[i].key);
```

Comment:
Since it was allocated in `(grn_table_sort_key *) palloc(sizeof(grn_table_sort_key) * nSortKeys)`, `grn_table_sort_key_close` could not be used.
Is there a better way?